src/datachain/catalog/catalog.py

```diff
@@ -1554,19 +1554,20 @@ def raise_termination_signal(sig: int, _: Any) -> NoReturn:
             raise TerminationSignal(sig)
 
         thread: Optional[Thread] = None
-        with subprocess.Popen(cmd, env=env, **popen_kwargs) as proc:  # noqa: S603
-            logger.info("Starting process %s", proc.pid)
 
-            orig_sigint_handler = signal.getsignal(signal.SIGINT)
-            # ignore SIGINT in the main process.
-            # In the terminal, SIGINTs are received by all the processes in
-            # the foreground process group, so the script will receive the signal too.
-            # (If we forward the signal to the child, it will receive it twice.)
-            signal.signal(signal.SIGINT, signal.SIG_IGN)
+        orig_sigint_handler = signal.getsignal(signal.SIGINT)
+        orig_sigterm_handler = signal.getsignal(signal.SIGTERM)
+        signal.signal(signal.SIGTERM, raise_termination_signal)
```

Comment:
We don't need to handle `TerminationSignal` to arrive in `subprocess.Popen`. If that happens before we `Popen` gets created, we'll bubble up the exception and there is no `popen`/`process` to shutdown.

Reply:
what happens if it comes right after `with subprocess.Popen(cmd, env=env, **popen_kwargs) as proc` ?

Reply:
In that case, from what I can tell, it will happen when running the code inside the `try .. except` block. (At the next bytecode).

But, could it happen within `subprocess.Popen.__enter__`? Probably, yes, it could, and `__exit__` would not ever be called.

See https://github.com/python/cpython/issues/75569.

Again, this signal handling is on the best effort basis. (Also this code does not really belong here, but on Studio, so I'd rather not complicate any further).

Reply:
Also found this: https://peps.python.org/pep-0419/#interruption-inside-with-statement-expression